packages/fx-core/src/plugins/resource/frontend/configs.ts

```diff
@@ -72,6 +72,9 @@ export class FrontendConfig {
     let storageName: string | undefined;
     if (isArmSupportEnabled()) {
       storageName = getArmOutput(ctx, ArmOutput.FrontendStorageName) as string;
+      if (!storageName) {
```

Comment:
The logic is quiet complex. Ok for now but we better to improve it.

Reply:
yes, I think we need to do some refractor to split the provision and deploy related logic here. Let's do this later.

Reply:
Suggest add "todo" comment to avoid being forgotten.